scipy/optimize/zeros.py

```diff
@@ -209,6 +211,93 @@ def newton(func, x0, fprime=None, args=(), tol=1.48e-8, maxiter=50,
     raise RuntimeError(msg)
 
 
+def _array_newton(func, x0, fprime, args, tol, maxiter, fprime2,
+                  failure_idx_flag=False):
+    """
+    A vectorized version of Newton, Halley, and secant methods for arrays. Do
+    not use this method directly. This method is called from :func:`newton`
+    when ``np.isscalar(x0)`` is true. For docstring, see :func:`newton`.
+    """
+    p0 = np.asarray(x0)  # convert to ndarray
+    failures = np.ones(p0.shape, dtype=bool)  # at start, nothing converged
```

Comment:
`failures` is a name that makes sense on exit, but in this code it's extremely misleading. I'd strongly recommend a more descriptive `active`, which would be `~failures`.

Reply:
I really like this, I had a though along the same lines, but didn't follow it ...